### ISSUE TITLE
P0 tests for a smoke test for every Rancher build

### ISSUE THREAD

**Issue by sowmyav27** [rancher/qa-tasks#487]
Tracking a List of tests for a smoke or a sanity check of Rancher, which could be either run every night or on every build. 
- [ ] RKE1 cluster prov
- [ ] RKE2 cluster prov
- [ ] Fleet p0 test
- [ ] _TBD_
- [ ] _TBD_

**To do**
- Add tests if we don't have ^^
- Have these specific tests run every night or on every build (_TBD_). 

--
Update:
09/28: Get feedback from Team what other tests can be added. (TBD when/who will work on this task)

**Comment by slickwarren** [comment]
(Work in progress)

if this is for every commit: 
* I think this list is good for now

if this is for nightly runs:
* add chart tests (deploy monitoring on rke1/rke2/k3s clusters)
* windows cluster / OS checks

**Comment by markusewalker** [comment]
Another option we can do is running these P0 tests twice a day as opposed to every Rancher commit. An example would be 12AM and 12PM or 3AM and 3PM (whatever we decide).

Regardless of this being every commit or nightly, we should have an automated test report that highlights what test failed and what test passed. This can be done with test rails and Jenkins or through other means, but it would be nice to have a summarized table so the team can take appropriate action quickly.

**Comment by slickwarren** [comment]
our go automation team also uses this tool for their summaries of pass/fail tests: https://github.com/gotestyourself/gotestsum

**Comment by slickwarren** [comment]
Now that I've been thinking about this with drone flakiness in mind, I'm no longer sure it would be a good idea on every commit. Like Markus said, maybe we could have some cron job (nightly, weekly, etc.) or maybe even every RC instead?

**Comment by markusewalker** [comment]
Coming back to this, @slickwarren I have an idea on how to implement this. See it below:

- Create a new Jenkins job, something like `go-nightly-runs`.
- This new job will be triggered daily in the exact same way that we do for OnTag, through a docker trigger directly calling the `rancher/rancher` repo
- Once my PR is merged for post-release Rancher checks, this job will directly call that job

The last bullet point needs a bit more exploring, but I think that can be the simplest way to get this going. What do you think?

**Comment by slickwarren** [comment]
I think that sounds good.

**Comment by markusewalker** [comment]
Marking this as completed. We have recurring Jenkins jobs for select tests to be ran daily, bi-weekly and weekly. Any further work can be isolated in separate issues.
